src/com/gfredericks/schpec/number.clj

```diff
@@ -0,0 +1,119 @@
+(ns com.gfredericks.schpec.number
+  (:require [clojure.spec :as s]
+            [clojure.spec.gen :as gen])
+  (:import [java.math BigDecimal MathContext RoundingMode]))
+
+(defn real?
+  [x]
+  (and (number? x)
+       (case x
+         Double/POSITIVE_INFINITY false
+         Double/NEGATIVE_INFINITY false
+         Double/NaN false
+         true)))
+
+(s/def ::real
+  (s/spec real?
+          :gen #(gen/double* {:infinite? false :NaN? false})))
+
+(defn decimal-pred
+  [precision scale]
+  (fn [d]
+    (let [d (bigdec d)]
```

Comment:
by doing this coercion are you aiming to let the spec match floating point numbers or integers? I usually prefer specs that lock down a specific type (like `BigDecimal` in this case) as it makes it easier to handle and test things when there aren't several possibilities.